Sources/TABResourceLoader/Protocols/Resource types/NetworkResourceType.swift

```diff
@@ -73,18 +76,26 @@ public protocol NetworkResourceType {
 // MARK: - NetworkJSONResource defaults
 public extension NetworkResourceType {
 
+  // MARK: Public properties
+
   public var httpRequestMethod: HTTPMethod { return .get }
   public var httpHeaderFields: [String: String]? { return [:] }
   public var jsonBody: Any? { return nil }
   public var queryItems: [URLQueryItem]? { return nil }
   public var requestTimeoutInterval: TimeInterval? { return nil }
+  public var urlQueryAllowedCharacterSet: CharacterSet { return .urlQueryAllowed }
+
+  // MARK: Public functions
 
-  public func urlRequest() -> URLRequest? {
-    var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: true)
-    let initialItems = urlComponents?.queryItems
-    urlComponents?.queryItems = allQueryItems(initialItems: initialItems)
+  func urlRequest() -> URLRequest? {
+    guard let urlComponents = createURLComponents() else {
+      return nil
+    }
+
+    guard let urlFromComponents = urlComponents.url else {
```

Comment:
Could you merge the above two `guards` into one? 

```swift
guard let urlFromComponents = createUrlComponents().url else { return nil }
```

Reply:
Done!